freshmaker/handlers/koji/rebuild_flatpak_application_on_module_ready.py

```diff
@@ -98,72 +119,219 @@ def _get_requests_session(cls, retry_options={}):
+        ) as session:
+            for image in images:
+                self.set_context(db_event)
+
+                image.resolve_commit()
+                image.resolve_original_odcs_compose_ids(False)
+                nvr = image.nvr
+                image_name = koji.parse_NVR(nvr)["name"]
+                build = self.record_build(
+                    self.event,
+                    image_name,
+                    ArtifactType.IMAGE,
+                    state=ArtifactBuildState.PLANNED.value,
+                    original_nvr=nvr,
+                    rebuild_reason=RebuildReason.DIRECTLY_AFFECTED.value,
+                )
+                # Set context to particular build so logging shows this build
+                # in case of error.
+                self.set_context(build)
+
+                module_name_stream_version_set = set()
+                module_nvrs = auto_build_image_modules_mapping[nvr]
+                for module_nvr in module_nvrs:
+                    mmd = session.get_modulemd(module_nvr)
+                    name = mmd.get_module_name()
+                    stream = mmd.get_stream_name()
+                    version = mmd.get_version()
+                    module_name_stream_version_set.add(
+                        ":".join([name, stream, str(version)])
+                    )
+                (
+                    renewed_odcs_compose_set,
+                    new_odcs_compose_set,
+                ) = self._prepare_data_for_compose(
+                    image["original_odcs_compose_ids"], module_name_stream_version_set
+                )
+
+                build.build_args = json.dumps(
+                    {
+                        "repository": image["repository"],
+                        "commit": image["commit"],
+                        "target": image["target"],
+                        "branch": image["git_branch"],
+                        "arches": image["arches"],
+                        "renewed_odcs_compose_ids": list(renewed_odcs_compose_set),
+                        "flatpak": image.get("flatpak", False),
+                        "isolated": image.get("isolated", True),
+                    }
+                )
+                db.session.commit()
+
+                for compose_source in new_odcs_compose_set:
+                    if compose_source in odcs_cache:
+                        db_compose = odcs_cache[compose_source]
+                    else:
+                        compose = create_odcs_client().new_compose(
+                            compose_source, "module"
+                        )
+                        db_compose = Compose(odcs_compose_id=compose["id"])
+                        db.session.add(db_compose)
+                        db.session.commit()
+                        odcs_cache[compose_source] = db_compose
+
+                    if db_compose:
+                        build.add_composes(db.session, [db_compose])
+                        db.session.commit()
```

Comment:
Can there be just a single DB commit at the end of this method?

Reply:
I am assuming after every db operation, it will do a commit. But I am not sure we will strictly following this.